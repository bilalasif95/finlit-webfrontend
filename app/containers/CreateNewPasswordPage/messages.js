/*
 * Create New Password Page Messages
 *
 * This contains all the text for the Create New Password Page.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.containers.CreateNewPasswordPage';

export default defineMessages({
  CreateNewPassword: {
    id: `${scope}.CreateNewPassword`,
    defaultMessage: 'Create New Password',
  },
  NewPassword: {
    id: `${scope}.NewPassword`,
    defaultMessage: 'New Password',
  },
  ConfirmPassword: {
    id: `${scope}.ConfirmPassword`,
    defaultMessage: 'Confirm Password',
  },
  PasswordMatch: {
    id: `${scope}.PasswordMatch`,
    defaultMessage: 'Both password must match',
  },
  ResetPassword: {
    id: `${scope}.ResetPassword`,
    defaultMessage: `Reset Password`,
  },
});
