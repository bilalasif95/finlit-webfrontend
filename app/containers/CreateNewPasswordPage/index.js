/*
 * Create New Password Page
 */
import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { Button, FormGroup, Label, Input } from 'reactstrap';
import messages from './messages';

export default function CreateNewPasswordPage() {
  return (
    <div className="registration_page">
      <Helmet>
        <title>- Create New Password</title>
        <meta name="description" content="FinLit - Create New Password Page" />
      </Helmet>
      <div className="form_container">
        <div className="form_content">
          <h2>
            <FormattedMessage {...messages.CreateNewPassword} />
          </h2>
          <div className="form">
            <FormGroup>
              <Label for="password">
                <FormattedMessage {...messages.NewPassword} />
              </Label>
              <Input
                type="password"
                name="newpassword"
                id="password"
                placeholder="New password"
              />
            </FormGroup>
            <FormGroup>
              <Label for="password">
                <FormattedMessage {...messages.ConfirmPassword} />
              </Label>
              <Input
                type="password"
                name="confirmpassword"
                id="password"
                placeholder="Confirm password"
              />
            </FormGroup>
            <div className="remember_forgot">
              <p>
                <FormattedMessage {...messages.PasswordMatch} />
              </p>
            </div>
            <Button>
              <FormattedMessage {...messages.ResetPassword} />
            </Button>
          </div>
        </div>
      </div>
      <div className="img_container" />
    </div>
  );
}
