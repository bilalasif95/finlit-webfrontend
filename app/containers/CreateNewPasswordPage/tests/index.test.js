import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { BrowserRouter } from 'react-router-dom';
import CreateNewPasswordPage from '../index';

describe('CreateNewPasswordPage />', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <BrowserRouter>
          <CreateNewPasswordPage />
        </BrowserRouter>
      </IntlProvider>,
    );

    expect(firstChild).toMatchSnapshot();
  });
});
