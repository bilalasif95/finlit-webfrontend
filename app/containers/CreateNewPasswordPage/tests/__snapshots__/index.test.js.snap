// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNewPasswordPage /> should render its heading 1`] = `
<div
  class="registration_page"
>
  <div
    class="form_container"
  >
    <div
      class="form_content"
    >
      <h2>
        <span>
          Create New Password
        </span>
      </h2>
      <div
        class="form"
      >
        <div
          class="form-group"
        >
          <label
            class=""
            for="password"
          >
            <span>
              New Password
            </span>
          </label>
          <input
            class="form-control"
            id="password"
            name="newpassword"
            placeholder="New password"
            type="password"
          />
        </div>
        <div
          class="form-group"
        >
          <label
            class=""
            for="password"
          >
            <span>
              Confirm Password
            </span>
          </label>
          <input
            class="form-control"
            id="password"
            name="confirmpassword"
            placeholder="Confirm password"
            type="password"
          />
        </div>
        <div
          class="remember_forgot"
        >
          <p>
            <span>
              Both password must match
            </span>
          </p>
        </div>
        <button
          class="btn btn-secondary"
        >
          <span>
            Reset Password
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="img_container"
  />
</div>
`;
